src/lib/Microsoft.Health.Extensions.Fhir.R4/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,59 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using EnsureThat;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Health.Extensions.Host.Auth;
+using Microsoft.Health.Logging.Telemetry;
+using FhirClient = Microsoft.Health.Fhir.Client.FhirClient;
+using IFhirClient = Microsoft.Health.Fhir.Client.IFhirClient;
+
+namespace Microsoft.Health.Extensions.Fhir
+{
+    public static class ServiceCollectionExtensions
+    {
+        public static void AddFhirClient(this IServiceCollection serviceCollection, IConfiguration configuration)
+        {
+            EnsureArg.IsNotNull(serviceCollection, nameof(serviceCollection));
+            EnsureArg.IsNotNull(configuration, nameof(configuration));
+
+            var url = new Uri(configuration.GetValue<string>("FhirService:Url"));
+            bool useManagedIdentity = configuration.GetValue<bool>("FhirClient:UseManagedIdentity");
+
+            serviceCollection.AddSingleton(typeof(ITelemetryLogger), typeof(IomtTelemetryLogger));
+            var serviceProvider = serviceCollection.BuildServiceProvider();
+            var logger = serviceProvider.GetRequiredService<ITelemetryLogger>();
+
+            serviceCollection.AddHttpClient<IFhirClient, FhirClient>(client =>
+            {
+                client.BaseAddress = url;
```

Comment:
>url [](http://example.com/codeflow?start=37&length=3)

More of a FYI, but since the url and logger are variables from an outside scope this means the entire contents of this method outside this lamda will be implicitly captured in a class in the compiled code.  

For more details and understanding the implications see here [https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/lambda-expressions#capture-of-outer-variables-and-variable-scope-in-lambda-expressions](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/lambda-expressions#capture-of-outer-variables-and-variable-scope-in-lambda-expressions)

Reply:
Thanks for sharing this. My understanding here is that the implicitly captured contents of the entire method outside the lambda will not affect the instantiation of the FhirClient. However, I may be incorrect. Happy to discuss further

Reply:
You should be fine. Nothing looked problematic in this code but I wanted to make you aware of the concept in principal.  Captured variables like this can artificially extend the life time of objects.